quolance-api/src/main/java/com/quolance/quolance_api/services/impl/UserServiceImpl.java

```diff
@@ -98,18 +97,15 @@ public void resetPassword(UpdateUserPasswordRequestDto request) {
 
     @Override
     @Transactional
-    public UserResponseDto updateUser(UpdateUserRequestDto request) {
-        User user = SecurityUtil.getAuthenticatedUser();
-        user = userRepository.getReferenceById(user.getId());
+    public UserResponseDto updateUser(UpdateUserRequestDto request, User user) {
         user.update(request);
         user = userRepository.save(user);
         return new UserResponseDto(user);
     }
 
     @Override
     @Transactional
-    public UserResponseDto updatePassword(UpdateUserPasswordRequestDto request) {
-        User user = SecurityUtil.getAuthenticatedUser();
```

Comment:
I dont think this should be changed. Only the user authenticated (within the security context) should be able to update his password, if you want to have this ability to update a user password by feeding the given user,it should be added on the adminService